go-controller/pkg/util/net.go

```diff
@@ -332,5 +329,5 @@ func IPNetsIPToStringSlice(ips []*net.IPNet) []string {
 // CalculateRouteTableID will calculate route table ID based on the network
 // interface index
 func CalculateRouteTableID(ifIndex int) int {
-	return ifIndex + routingTableIDStart
+	return math.MaxUint32 - ifIndex
```

Comment:
hmm why is that? what conflicts do you mean?

Reply:
Users are more likely to create tables at lower integer ranges i think without reading documentation and conflict with tables we create.

Reply:
I havent seen any cases where this has happened yet with EIP but guessing.

Reply:
hmm I think documenting it should be enough? but given you have done the work here already I don't mind going with this....ultimately I get this feeling someone will say they want to make it configurable LOL!

Reply:
I dont mind discarding it if its not worth it. No problem. I think its sensible not to depend on docs alone and this allows us to avoid human readable route tables integers.

Reply:
I agree with starting higher, but I have 2 concerns:
1. It looks like you are starting from the max and decrementing. The maximum is 4294967295. This is a really big number for a user or a developer to just memorize as they are trying to debug and look at routes in their UDN route table. I wonder if there is some kind of compromise where we could start at like 10000 or something?
2. By changing this, does this break upgrade for egress IP that is currently using some other table? Or how does that work? Will there be outage or is it only for assigning new table numbers?

Reply:
> 2. By changing this, does this break upgrade for egress IP that is currently using some other table? Or how does that work? Will there be outage or is it only for assigning new table numbers?

I think no because when upgrading during node reboots we will start fresh. so the new tables would get picked up on their own based on the new ifaceindex

Reply:
> I dont mind discarding it if its not worth it. No problem. I think its sensible not to depend on docs alone and this allows us to avoid human readable route tables integers.

no I don't mean its not worth it.. the change makes sense... just that we need the PRs that are higher priority in before this :D + we still need to document the range we use just to be sure.

Reply:
Maybe I didn't phrase my question right or I'm misunderstanding something. We upgrade ovnk-node first (before reboot). So my question is, lets say today we are using VRF 5 (just randomly picking a number) for egress IP. Now OVNK upgrades, will it ocntinue using VRF 5 and there is no outage, or will it now pick a new VRF table and reprogram it due to this commit?

Reply:
I think its possible theres disruption when switching to the new routing table.

Reply:
but I am not sure and probably should write a test.